### PR TITLE
Update vault-as-secret-store.md with plugin changes

### PR DESCRIPTION
**Pull request by ezgidemirel**
This PR updates existing Vault as an ESS guide with plugin change.

### CODE REVIEW

File: content/knowledge-base/integrations/vault-as-secret-store.md [hunk 1]
```diff
@@ -153,34 +154,23 @@ kubectl -n vault-system exec -it vault-0 -- vault write auth/kubernetes/role/cro
 
 ### Install and Configure Crossplane
 
-1. Install Crossplane by:
+> Prerequisite: Plugin support will be available in Crossplane v1.13.0. In this doc, we
+> will use the release candidate version of Crossplane 1.12.0-rc.0.284.g190ab067.
```
Comment (lsviben) [inline content/knowledge-base/integrations/vault-as-secret-store.md:158]: We plan to update this when Crossplane v1.13.0 is released?

**Review by lsviben — APPROVED** [review]
Added a nonblocking comment/question. 

Went through the whole guide and got the expected results. Great stuff, easy to follow.

File: content/knowledge-base/integrations/vault-as-secret-store.md (hunk 1 above)
Comment (plumbis) [reply content/knowledge-base/integrations/vault-as-secret-store.md:158]: Yes, we will update it. I opened [C/C #3995](https://github.com/crossplane/crossplane/issues/3995) to make sure I update it after v1.13 is released.

**Review by plumbis — CHANGES_REQUESTED** [review]
@ezgidemirel I submitted a PR to your branch with my suggested edits. 


https://github.com/ezgidemirel/docs/pull/1

File: content/knowledge-base/integrations/vault-as-secret-store.md (hunk 1 above)
Comment (phisco) [reply content/knowledge-base/integrations/vault-as-secret-store.md:158]: Is this 1.13 or 1.12? 🤔

Comment (ezgidemirel) [reply content/knowledge-base/integrations/vault-as-secret-store.md:158]: It's 1.12

**Pull request closed without merging**